### ISSUE TITLE
pdf() device transparency problems with shape = "outline"

### ISSUE THREAD

**Issue by hughjonesd** [hughjonesd/ggmagnify#23]
I just tested the "Texas" commands and I see that it works for PDF but not PNG Device.

However, while the transparency of the backbround of the PDF version is achieved, the Texas state itself is not colored red. With `shadow = TRUE` there is a gray coloring inside. With `shadow = FALSE` there is only the orange outline. I colored the rest of the states "green4" with "yellow" lines to make the image different. Here are the PDF (Device) output using `pdf(file="Texasplot2.pdf")` and `dev.off()`:

<img width="1373" alt="texas2" src="https://github.com/hughjonesd/ggmagnify/assets/5881768/376b1684-b9e7-46f5-85eb-740367d69c4c">

(Saving to PNG looks the same as the one on the first comment, with a red Texas but blue neighbours.)

_Originally posted by @jsgro in https://github.com/hughjonesd/ggmagnify/issues/22#issuecomment-1920038263_

**Comment by hughjonesd** [comment]
Bug checks out on my machine too. Using `bg = "white"` in the `pdf` call doesn't fix it.

**Comment by jsgro** [comment]
How did you create the Texas image that is on the Code section at
https://github.com/hughjonesd/ggmagnify#maps-experimental

It seems that "it used to work"...

**Comment by hughjonesd** [comment]
Why do you think that was created with a pdf device?

**Comment by jsgro** [comment]
My question was "How did you create the Texas image" - did you use another "graphics device" other than PDF?

I have no other information. 
Perhaps are you on a Linux system? A newer OS? a different Windows?

**Comment by hughjonesd** [comment]
I understand, but this is an issue tracker, not a help forum. Thanks for the bug report. I'm not sure what is going on, but will look at it.

**Comment by hughjonesd** [comment]
`shape = "rect"` and `"ellipse"` both work OK with PDF, so it must be specific to the `"outline"`.

**Comment by jsgro** [comment]
There are always many variables. R version, hardware etc.
I do not know how your "red Texas" inset worked when you created this image, but glad that it is a "real bug" that was found.
Best. JYS

**Comment by hughjonesd** [comment]
The problem is here:
```
    vp <- viewport(x = mean(x_rng), y = mean(y_rng), width = diff(x_rng),
                   height = diff(y_rng), default.units = "native",
                   mask = mask_grob)
    plot_gtable <- grid::editGrob(plot_gtable, vp = vp)
```

With pdf, plot_gtable becomes invisible via grid.draw() after the editGrob call.

**Comment by hughjonesd** [comment]
The problem is *not* border_grob replacing the colour with transparency. Seems to be to do with the mask grob; setting mask_grob = FALSE works fine but then obviously leaves a square inset.

**Comment by hughjonesd** [comment]
A workaround on Mac is to use the `cairo_pdf()` device, which works nicely. That may also work on other platforms where it is supported.

**Comment by hughjonesd** [comment]
Added documentation on recommended graphics devices in the help file. Leaving this open for others to find.

**Comment by jsgro** [comment]
Thanks for keeping this open... Glad you added the paragraph `Graphics devices` under `Tips and tricks`.  I followed the information and here are suggestions to clarify the text:

- the package is called **`ragg`** (`install.packages('ragg')` as trying to install **`ragg_png`** causes error.)
- the command is not `ragg_png` but  **`agg_png`** as I found in the `ragg` package [README.html](https://cran.r-project.org/web/packages/ragg/readme/README.html)

I think that having this information at the Top of the `README` file would be useful as other people might not realize `agg_png` was the device used. While this creates the best image, most users would not know about it... 

I see that there are other options within that package for other formats and has great features (speed, font access etc.) Well worth the discovery! Thanks. JYS

**Comment by hughjonesd** [comment]
Yes, but from knitr the device option must be set to "ragg_png" (and no, I don't know why this name was chosen...) See README.Rmd source for an example.

**Comment by hughjonesd** [comment]
You can't hold people's hands too much. Otherwise, you just end up reproducing other packages' documentation. Then you have increased the burden on yourself, and when the other package updates, your docs are inaccurate. It's fine to expect people to figure out agg_png/ragg_png by themselves.

**Comment by hughjonesd** [comment]
Also again, no offence, but this is an issue tracker, not a forum. I didn't notice till now, but this question is completely irrelevant to this issue. That makes it harder for me to track the state of play for the actual bug. Please keep issues separate.
